### PR TITLE
Matches history

### PR DESCRIPTION
**Pull request by moralesbang**
## Summary

- Added matches history table.

## Screenshots

Screen Shot 2019-08-10 at 3 21 17 PM

## Trello Card

https://trello.com/c/BmDlan80

### CODE REVIEW

File: react/tic-tac-toe-api/package.json [hunk 1]
```diff
@@ -10,5 +10,6 @@
   "license": "MIT",
   "dependencies": {
     "json-server": "^0.14.0"
-  }
+  },
```
Comment (alejobh) [inline react/tic-tac-toe-api/package.json]: remove this line and below line

File: react/tic-tac-toe/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": ["wolox-react"],
+  "rules": {
+    "no-magic-numbers": 0,
+    "class-methods-use-this": 0
```
Comment (alejobh) [inline react/tic-tac-toe/.eslintrc]: why this disable?

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  componentDidMount() {
+    services.getMatches().then(response => this.setState({ matches: response.data }));
+  }
+
+
+  humanize = (property) => property.replace(/_/g, ' ')
+    .replace(/(\w+)/g, (match) => match.charAt(0).toUpperCase() + match.slice(1));
+
+  MATCH_RESULT_CLASS = {
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: This is not used

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  state = {
+    matches: []
+  }
+
+  componentDidMount() {
+    services.getMatches().then(response => this.setState({ matches: response.data }));
+  }
+
+
+  humanize = (property) => property.replace(/_/g, ' ')
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: This is more workload than you need, just use a dictionary por matching the correct player or Tie

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+import styles from './styles.module.scss';
+
+class MatchHistory extends Component {
+  state = {
+    matches: []
+  }
+
+  componentDidMount() {
+    services.getMatches().then(response => this.setState({ matches: response.data }));
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: Create a function that checks the loading state, the error state and the matches

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+
+import services from '~services/MatchesService'; // eslint-disable-line import/no-unresolved
+
+import styles from './styles.module.scss';
+
+class MatchHistory extends Component {
+  state = {
+    matches: []
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: What about `Loading` and `Error` states?

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        <thead>
+          <tr className={styles.headerRow}>
+            <th className={styles.headerCell}>Player 1</th>
+            <th className={styles.headerCell}>Player 2</th>
+            <th className={styles.headerCell}>Winner</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {matches.map(match => (
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: If matches is `[]` you could display a message to the user that there is no matches.

File: react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+const PLAYER_IDS = {
+  playerOne: 'player_one',
+  playerTwo: 'player_two'
+};
+
+const WINNERS = {
+  [PLAYER_IDS.playerOne]: 'Player One',
+  [PLAYER_IDS.playerTwo]: 'Player Two',
+  tie: 'Tie'
+};
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js]: `PLAYER_IDS` and `WINNERS ` should be on a const file

**Pull request merged**